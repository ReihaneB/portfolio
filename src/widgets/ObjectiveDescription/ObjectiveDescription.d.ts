export interface ObjectiveDescriptionProps {
  title: string;
  description: string;
}
