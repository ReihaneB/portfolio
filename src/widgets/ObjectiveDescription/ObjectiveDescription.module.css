.root {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: var(--blue-lighter);
  border-radius: 16px;
  padding: 32px;

  @media (width <= 768px) {
    padding: 24px;
  }
}

.header {
  width: fit-content;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: var(--white);
  border-radius: 9999px;
  padding: 16px 32px;
}

.icon {
  margin-right: 12px;
}

.content {
  margin-top: 32px;
  align-self: flex-start;

  @media (width <= 768px) {
    margin-top: 24px;
  }
}

.title {
  margin-bottom: 16px;
}