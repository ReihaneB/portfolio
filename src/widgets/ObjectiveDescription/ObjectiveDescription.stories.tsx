import type { Meta, StoryObj } from '@storybook/react';

import ObjectiveDescription from './ObjectiveDescription';

const meta: Meta<typeof ObjectiveDescription> = {
  title: 'Widgets/ObjectiveDescription',
  component: ObjectiveDescription,
};

export default meta;
type Story = StoryObj<typeof ObjectiveDescription>;

export const Playground: Story = {
  args: {
    title: 'Dolore aliquip dolor consectetur esse eu non. Adipisicing cupidatat cillum mollit sunt cupidatat cupidatat voluptate ex laborum aute irure.',
    description: 'Aliqua sint amet reprehenderit commodo. Nisi duis esse incididunt non ad eiusmod labore sunt enim sunt. Anim est nisi veniam veniam ipsum mollit dolor. Labore ad sunt excepteur sunt velit id cillum eiusmod ut cupidatat adipisicing. Tempor ex anim mollit elit adipisicing reprehenderit eiusmod sunt. Ipsum minim deserunt ea culpa aute fugiat ut quis ullamco proident non pariatur et ex.',
  },
};

export const LongDescription: Story = {
  args: {
    title: 'Nisi labore cupidatat esse Lorem dolore quis anim occaecat non culpa.',
    description: 'Non reprehenderit irure ex occaecat duis nulla officia dolore magna Lorem duis. Dolore culpa aute aliquip sint eu irure aute mollit aliqua. Officia dolore et laboris proident esse deserunt irure minim magna ad ut sint. Incididunt elit Lorem qui cillum dolore duis labore laborum consequat minim. Veniam excepteur incididunt ad labore. Sint pariatur irure consequat id esse anim esse qui. Est adipisicing sit irure aliqua sint culpa nostrud ut laboris reprehenderit Lorem pariatur eiusmod cillum. Cillum magna mollit minim non proident dolor mollit aliquip eu voluptate duis aliquip voluptate. Officia reprehenderit officia amet anim. Magna officia quis aliqua officia anim pariatur officia sit esse laborum. Cupidatat cupidatat id incididunt officia et id cillum. Sint reprehenderit consectetur aute do enim do exercitation ex adipisicing ipsum proident magna consequat. Culpa dolore commodo ullamco occaecat aute cillum cupidatat ad ipsum sint non ea laboris magna.',
  },
};

export const ShortDescription: Story = {
  args: {
    title: 'Ut ut ipsum dolor quis id sit fugiat cupidatat sit Lorem laborum ut. Eiusmod proident laborum est mollit in cillum id do proident cupidatat anim enim. Elit ut anim eu reprehenderit. Consequat commodo irure adipisicing in in quis.',
    description: 'Dolor qui amet Lorem aliquip elit sit anim. Adipisicing cupidatat cillum mollit sunt cupidatat cupidatat voluptate ex laborum aute irure. Dolore sit cillum anim eiusmod sit quis dolore id id cupidatat laborum culpa.',
  },
};
