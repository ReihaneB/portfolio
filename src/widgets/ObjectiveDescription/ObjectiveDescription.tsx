import NextImage from 'next/image';

import IconGoal from '@/styles/icons/meta/goal.svg';

import type { ObjectiveDescriptionProps } from './ObjectiveDescription.d';
import styles from './ObjectiveDescription.module.css';


function ObjectiveDescription({
  title,
  description,
  ...rest
}: ObjectiveDescriptionProps) {
  return (
    <div
      className={styles.root}
      {...rest}
    >
      <div className={styles.header}>
        <NextImage
          src={IconGoal}
          alt="Un icÃ´ne de cible pour exprimer l'objectif du projet."
          width={24}
          height={24}
          className={styles.icon}
        />
        <h2>L'objectif</h2>
      </div>
      <div className={styles.content}>
        <h3 className={styles.title}>
          {title}
        </h3>
        <p className={styles.description}>
          {description}
        </p>
      </div>
    </div>
  );
}

export default ObjectiveDescription;
