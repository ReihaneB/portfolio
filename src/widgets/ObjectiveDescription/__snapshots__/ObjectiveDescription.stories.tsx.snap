// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets/ObjectiveDescription LongDescription smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__root--yDE6T">
    <div class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__header--WmvFD">
      <img alt="Un icône de cible pour exprimer l'objectif du projet."
           loading="lazy"
           width="24"
           height="24"
           decoding="async"
           data-nimg="1"
           class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__icon--ynsEP"
           srcset="/static/media/goal.9261bfa0.svg?w=32&amp;q=75 1x, /static/media/goal.9261bfa0.svg?w=48&amp;q=75 2x"
           src="/static/media/goal.9261bfa0.svg?w=48&amp;q=75"
           style="color: transparent;"
      >
      <h2>
        L'objectif
      </h2>
    </div>
    <div class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__content--P025e">
      <h3 class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__title--ORMcl">
        Nisi labore cupidatat esse Lorem dolore quis anim occaecat non culpa.
      </h3>
      <p>
        Non reprehenderit irure ex occaecat duis nulla officia dolore magna Lorem duis. Dolore culpa aute aliquip sint eu irure aute mollit aliqua. Officia dolore et laboris proident esse deserunt irure minim magna ad ut sint. Incididunt elit Lorem qui cillum dolore duis labore laborum consequat minim. Veniam excepteur incididunt ad labore. Sint pariatur irure consequat id esse anim esse qui. Est adipisicing sit irure aliqua sint culpa nostrud ut laboris reprehenderit Lorem pariatur eiusmod cillum. Cillum magna mollit minim non proident dolor mollit aliquip eu voluptate duis aliquip voluptate. Officia reprehenderit officia amet anim. Magna officia quis aliqua officia anim pariatur officia sit esse laborum. Cupidatat cupidatat id incididunt officia et id cillum. Sint reprehenderit consectetur aute do enim do exercitation ex adipisicing ipsum proident magna consequat. Culpa dolore commodo ullamco occaecat aute cillum cupidatat ad ipsum sint non ea laboris magna.
      </p>
    </div>
  </div>
</main>
`;

exports[`Widgets/ObjectiveDescription Playground smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__root--yDE6T">
    <div class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__header--WmvFD">
      <img alt="Un icône de cible pour exprimer l'objectif du projet."
           loading="lazy"
           width="24"
           height="24"
           decoding="async"
           data-nimg="1"
           class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__icon--ynsEP"
           srcset="/static/media/goal.9261bfa0.svg?w=32&amp;q=75 1x, /static/media/goal.9261bfa0.svg?w=48&amp;q=75 2x"
           src="/static/media/goal.9261bfa0.svg?w=48&amp;q=75"
           style="color: transparent;"
      >
      <h2>
        L'objectif
      </h2>
    </div>
    <div class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__content--P025e">
      <h3 class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__title--ORMcl">
        Dolore aliquip dolor consectetur esse eu non. Adipisicing cupidatat cillum mollit sunt cupidatat cupidatat voluptate ex laborum aute irure.
      </h3>
      <p>
        Aliqua sint amet reprehenderit commodo. Nisi duis esse incididunt non ad eiusmod labore sunt enim sunt. Anim est nisi veniam veniam ipsum mollit dolor. Labore ad sunt excepteur sunt velit id cillum eiusmod ut cupidatat adipisicing. Tempor ex anim mollit elit adipisicing reprehenderit eiusmod sunt. Ipsum minim deserunt ea culpa aute fugiat ut quis ullamco proident non pariatur et ex.
      </p>
    </div>
  </div>
</main>
`;

exports[`Widgets/ObjectiveDescription ShortDescription smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__root--yDE6T">
    <div class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__header--WmvFD">
      <img alt="Un icône de cible pour exprimer l'objectif du projet."
           loading="lazy"
           width="24"
           height="24"
           decoding="async"
           data-nimg="1"
           class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__icon--ynsEP"
           srcset="/static/media/goal.9261bfa0.svg?w=32&amp;q=75 1x, /static/media/goal.9261bfa0.svg?w=48&amp;q=75 2x"
           src="/static/media/goal.9261bfa0.svg?w=48&amp;q=75"
           style="color: transparent;"
      >
      <h2>
        L'objectif
      </h2>
    </div>
    <div class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__content--P025e">
      <h3 class="src-widgets-ObjectiveDescription-ObjectiveDescription-module__title--ORMcl">
        Ut ut ipsum dolor quis id sit fugiat cupidatat sit Lorem laborum ut. Eiusmod proident laborum est mollit in cillum id do proident cupidatat anim enim. Elit ut anim eu reprehenderit. Consequat commodo irure adipisicing in in quis.
      </h3>
      <p>
        Dolor qui amet Lorem aliquip elit sit anim. Adipisicing cupidatat cillum mollit sunt cupidatat cupidatat voluptate ex laborum aute irure. Dolore sit cillum anim eiusmod sit quis dolore id id cupidatat laborum culpa.
      </p>
    </div>
  </div>
</main>
`;
